### ISSUE TITLE
Script Engine Server Crash

### ISSUE THREAD

**Issue by Nylux** [DDDDDragon/ModShardLauncher#39]
## Description:

Attempting to create a new Script Engine Server using the UI button causes MSL to crash.

## Expected Behavior:

MSL should display a dialog saying the Script Engine Server is already running.

## Actual Behavior:

MSL attempts to create a new Socket connection with the exact same address and crashes.

## Additional Context:

N/A

## Possible Solution:

Check if the Script Engine Server is running, and if so, display a dialog saying the Script Engine Server is already running instead of trying to create a new one which causes a crash.

**Comment by DDDDDragon** [comment]
every time you close the window of server, i make the server end
so it's just maybe you dont close it and try to reopen a new one

**Comment by Nylux** [comment]
That's exactly it.
If you keep it open and try to open another one, MSL crashes.

That's not great, it's a mistake users could make and it would be better if we handle that possibility.

**Comment by DDDDDragon** [comment]
hmmm alright

**Comment by Nylux** [comment]
e825c33 was a great change but now we have a crash when trying to close the Script Engine !

```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.                                                                                
   at ModShardLauncher.Mods.ModInterfaceServer.<>c.<StartServer>b__9_1(IAsyncResult ar) in B:\WORK\MODDING\stoneshard-mods\Tools\MSL\ModShardLauncher\Mods\ModInterfaceEngine.cs:line 158
   at System.Threading.Tasks.TaskToApm.TaskAsyncResult.InvokeCallback()                                                                                                                  
   at System.Threading.Tasks.AwaitTaskContinuation.<>c.<.cctor>b__17_0(Object state)                                                                                                     
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)                                                           
--- End of stack trace from previous location ---                                                                                                                                        
   at System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, Object state, Task& currentTask)                                                                
--- End of stack trace from previous location ---                                                                                                                                        
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)                                                                                                                 
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()                                                                                                                 
   at System.Threading.ThreadPoolWorkQueue.Dispatch()                                                                                                                                    
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()                                                                                                               
   at System.Threading.Thread.StartCallback()
```

**Comment by DDDDDragon** [comment]
fix it

**Comment by Nylux** [comment]
Confirmed closed by commit 9324d68.
